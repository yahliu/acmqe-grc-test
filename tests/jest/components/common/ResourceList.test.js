/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import React from 'react'
import ResourceFilterBar from '../../../../src-web/components/common/ResourceFilterBar'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import * as reducers from '../../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import GrcApolloClient from '../../../../src-web/utils/client/apollo-client'
import { ApolloProvider } from '@apollo/client/react'
import { Provider } from 'react-redux'
import { updateActiveFilters } from '../../../../src-web/actions/common'

describe('ResourceFilterBar no filters applied', () => {
  it('renders as expected', () => {
    const history = {
      'length': 50,
      'action': 'POP',
      'location': {
        'pathname': '/multicloud/policies/',
        'search': '',
        'hash': ''
      }
    }
    const location = {
      'pathname': '/multicloud/policies/',
      'search': '',
      'hash': ''
    }
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = renderer.create(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <Provider store={store}>
          <BrowserRouter>
            <ResourceFilterBar
              activeFilters={{}}
              history={history}
              location={location}
              updateActiveFilters={updateActiveFilters}
            />
          </BrowserRouter>
        </Provider>
      </ApolloProvider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('ResourceFilterBar w/ filter applied', () => {
  it('renders as expected', () => {
    const activeFilters = {
      'standards': [
        'HIPAA'
      ]
    }
    const history = {
      'length': 50,
      'action': 'POP',
      'location': {
        'pathname': '/multicloud/policies/',
        'search': '',
        'hash': ''
      }
    }
    const location = {
      'pathname': '/multicloud/policies/',
      'search': '',
      'hash': ''
    }
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component2 = renderer.create(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <Provider store={store}>
          <BrowserRouter>
            <ResourceFilterBar
              activeFilters={activeFilters}
              history={history}
              location={location}
              updateActiveFilters={updateActiveFilters}
            />
          </BrowserRouter>
        </Provider>
      </ApolloProvider>
    )
    expect(component2.toJSON()).toMatchSnapshot()
  })
})
