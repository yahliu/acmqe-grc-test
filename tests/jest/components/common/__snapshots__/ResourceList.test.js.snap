// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceFilterBar no filters applied renders as expected 1`] = `null`;

exports[`ResourceFilterBar w/ filter applied renders as expected 1`] = `null`;
